File: 2126-destroying-asteroids/2126-destroying-asteroids.js
/**
 * @param {number} mass
 * @param {number[]} asteroids
 * @return {boolean}
 */
var asteroidsDestroyed = function(mass, asteroids) {
    asteroids.sort((a, b) => Number(a) - Number(b));
    for(let a of asteroids) {
        if(mass < a) return false;
        mass = mass + a;
    }
    return true;
};

File: 2126-destroying-asteroids/2126-destroying-asteroids.java
class Solution {
    public boolean asteroidsDestroyed(int mass, int[] asteroids) {
        Arrays.sort(asteroids);
        long tm = mass;
        for(int a : asteroids) {
            if(a > tm) return false;
            tm += a;
        }
        return true;
    }
}